prover/crates/lib/prover_job_processor/src/job_picker.rs

```diff
@@ -0,0 +1,18 @@
+use async_trait::async_trait;
+
+use crate::Executor;
+
+/// Job Picker trait, in charge of getting a new job for executor.
+/// NOTE: Job Pickers are tied to an executor, which ensures input/output/metadata types match.
+#[async_trait]
+pub trait JobPicker: Send + Sync + 'static {
+    type ExecutorType: Executor;
```

Comment:
I wonder why this is parametrized with `ExecutorType` - does it need to know what the executor type is? Seems like it only needs the Input and Metadata types